src/coord/src/coord.rs

```diff
@@ -7,33 +7,67 @@
+//! timestamps, and transactions.
 //!
-//! The command coordinator maintains a view of the installed
-//! views, and for each tracks the frontier of available times
-//! ([`upper`](arrangement_state::Frontiers::upper)) and the frontier
-//! of compacted times ([`since`](arrangement_state::Frontiers::since)).
-//! The upper frontier describes times that may not return immediately, as any
-//! timestamps in advance of the frontier are still open. The since frontier
-//! constrains those times for which the maintained view will be correct,
-//! as any timestamps in advance of the frontier must accumulate to the same
-//! value as would an un-compacted trace. The since frontier cannot be directly
-//! mutated, but instead can have multiple handles to it which forward changes
-//! from an internal MutableAntichain to the since.
+//! The command coordinator maintains a view of the installed views, and for
+//! each tracks the frontier of available times
+//! ([`upper`](arrangement_state::Frontiers::upper)) and the frontier of
+//! compacted times ([`since`](arrangement_state::Frontiers::since)). The upper
+//! frontier describes times that may not return immediately, as any timestamps
+//! in advance of the frontier are still open. The since frontier constrains
+//! those times for which the maintained view will be correct, as any
+//! timestamps in advance of the frontier must accumulate to the same value as
+//! would an un-compacted trace. The since frontier cannot be directly mutated,
+//! but instead can have multiple handles to it which forward changes from an
+//! internal MutableAntichain to the since.
+//!
+//! The [`Coordinator`] tracks various compaction frontiers so that indexes,
+//! compaction, and transactions can work together.
+//! [`determine_timestamp()`](Coordinator::determine_timestamp) returns the
+//! least valid since of its sources. Any new transactions should thus always
+//! be >= the current compaction frontier and so should never change the
+//! frontier when being added to [`txn_reads`](Coordinator::txn_reads). The
+//! compaction frontier may change when a transaction ends (if it was the
+//! oldest transaction and the index's since was advanced after the transaction
+//! started) or when [`update_upper()`](Coordinator::update_upper) is run (if
+//! there are no in progress transactions before the new since). When it does,
+//! it is added to [`since_updates`](Coordinator::since_updates) and will be
+//! processed during the next [`maintenance()`](Coordinator::maintenance) call.
+//!
+//! ## Frontiers another way
+//!
+//! If the above description of frontiers left you with questions, this
+//! repackaged explanation might help.
+//!
+//! - `upper` is the most recent time that you can read from sources. Reads at
+//!   times greater than `upper` may not return because the answer simply does
+//!   not exist yet.
+//!
+//!   For latency's sake, the coordinator will never choose to read at a time
```

Comment:
This isn't true. We choose times in advance of the upper for any read of tables, or in the case of a query over multiple indexes, we may choose a timestamp after some upper because a since of another index is ahead of that upper. I'm also a bit wary of using the word latency here, because it could mean two things (latency out of mz, or latency between a source and a result). Freshness is a word we've used internally to mean "most recent results possible without waiting". Maybe:

> The coordinator will try to use an arrangement's `upper` to return the freshest results without waiting.

I think it's important to not use the phrases "will use" or "require" here, because this isn't a correctness requirement. "try" weakens this.